ci/Jenkinsfiles/export.groovy

```diff
@@ -0,0 +1,255 @@
+  [$class: 'GithubProjectProperty', projectUrlStr: 'https://github.com/nuxeo/nuxeo-explorer/'],
+  [$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', daysToKeepStr: '60', numToKeepStr: '60', artifactNumToKeepStr: '5']],
+  disableConcurrentBuilds(),
+])
+
+String getCommitSha1() {
+  return sh(returnStdout: true, script: 'git rev-parse HEAD').trim();
+}
+
+pipeline {
+
+  agent {
+    label 'jenkins-nuxeo-package-11'
+  }
+
+  parameters {
+    string(name: 'BRANCH_NAME', defaultValue: 'master', description: 'The branch holding explorer version to be used (can also match a GitHub tag)')
+    string(name: 'NUXEO_IMAGE_VERSION', defaultValue: '', description: 'Version of the target Nuxeo Server Image')
+    booleanParam(name: 'USE_REFERENCE_PACKAGE_LIST', defaultValue: false, description: 'Install reference packages')
+    choice(name: 'CONNECT_URL', choices: ['https://connect.nuxeo.com/nuxeo', 'https://nos-preprod-connect.nuxeocloud.com/nuxeo'], description: 'Target Connect instance where packages will be retrieved')
+    text(name: 'PACKAGE_LIST', defaultValue: '', description: 'The list of packages to install for snapshot. Example: nuxeo-jsf-ui nuxeo-quota-1.0.0')
+    text(name: 'CONNECT_CLID', defaultValue: '', description: 'CLID to be used to install packages (a default one will be used if unset)')
+    string(name: 'SNAPSHOT_NAME', defaultValue: 'Nuxeo Platform', description: 'Name of the distributionn that will be exported')
+    booleanParam(name: 'UPLOAD_SNAPSHOT', defaultValue: true, description: 'Upload snapshot to target Explorer site (snapshot will also be archived)')
+    string(name: 'EXPLORER_URL', defaultValue: 'https://preview-nuxeo-explorer-master.platform.dev.nuxeo.com/nuxeo', description: 'Target Explorer site URL')
+    string(name: 'EXPLORER_AUTH_USERNAME', defaultValue: 'apidocAdmin', description: 'Target Explorer site username')
+    string(name: 'EXPLORER_AUTH_TOKEN', defaultValue: 'https://preview-nuxeo-explorer-master.platform.dev.nuxeo.com/nuxeo', description: 'Target Explorer site authentication token')
+  }
+
+  environment {
+    MAVEN_OPTS = "$MAVEN_OPTS -Xms512m -Xmx3072m"
+    MAVEN_ARGS = '-B -nsu'
+    REFERENCE_BRANCH = 'master'
+    SCM_REF = "${getCommitSha1()}"
+
+    // override templates to avoid activating packages at startup
+    NUXEO_TEMPLATE_OVERRIDE = "${BRANCH_NAME == REFERENCE_BRANCH ? '' : 'nuxeo.templates=default'}"
+
+    NUXEO_DOCKER_REGISTRY = 'docker-private.packages.nuxeo.com'
+    NUXEO_IMAGE_VERSION = getNuxeoImageVersion()
+    PREVIEW_NAMESPACE = "$APP_NAME-${BRANCH_NAME.toLowerCase()}"
+    ORG = 'nuxeo'
+  }
+
+  environment {
+    APP_NAME = "nuxeo-explorer-export"
+    NUXEO_IMAGE_VERSION = getNuxeoVersion(params.NUXEO_VERSION)
+    NUXEO_DOCKER_REGISTRY = 'docker-private.packages.nuxeo.com'
+    ORG = 'nuxeo'
+    VERSION = "${RELEASE_VERSION}"
+    NUXEO_IMAGE_NAME = 'nuxeo'
+    DRY_RUN = "${params.DRY_RUN}"
+    BRANCH_NAME = "${params.BRANCH_NAME}"
+  }
```

Comment:
is there a reason to make 2 separate environment directives ?

Reply:
No this is a very very (i mean, very) early draft version where i copy/pasted a bunch of things.
I created the PR early to check the build is still going through as i makes changes.
Your review will be much appreciated a bit later :)